packages/js/components/src/experimental-select-control/async-select-control.tsx

```diff
@@ -0,0 +1,168 @@
+/**
+ * External dependencies
+ */
+import { Spinner } from '@wordpress/components';
+import {
+	createElement,
+	useState,
+	useMemo,
+	useCallback,
+	useEffect,
+} from '@wordpress/element';
+import { debounce } from 'lodash';
+import { ReactElement, Component } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import { SelectControlProps, SelectControl } from './select-control';
+import { Menu } from './menu';
+import { MenuItem } from './menu-item';
+import {
+	defaultGetFilteredItems,
+	defaultGetItemLabel,
+	defaultGetItemValue,
+} from './utils';
+import { ChildrenProps } from './types';
+
+type AsyncSelectControlProps< ItemType > = Omit<
+	SelectControlProps< ItemType >,
+	'onInputChange' | 'items' | 'children'
+> & {
+	onSearch: ( value: string | undefined ) => Promise< ItemType[] >;
```

Comment:
Why `| undefined`? Why would we ever call `onSearch` with `value === undefined`?